src/bindings/js/node/src/tensor.cpp

```diff
@@ -191,3 +192,42 @@ Napi::Value TensorWrap::is_continuous(const Napi::CallbackInfo& info) {
     }
     return Napi::Boolean::New(env, _tensor.is_continuous());
 }
+
+void TensorWrap::set_shape(const Napi::CallbackInfo& info) {
+    Napi::Env env = info.Env();
+
+    if (!info[0].IsArray()) {
+        Napi::TypeError::New(env, "The argument to setShape() must be an array.").ThrowAsJavaScriptException();
+        return;
+    }
+
+    ov::Shape shape;
+    try {
+        shape = js_to_cpp<ov::Shape>(info, 0);
+    } catch (const std::exception& e) {
+        Napi::TypeError::New(env, std::string("Invalid shape parameter: ") + e.what()).ThrowAsJavaScriptException();
+        return;
+    }
```

Comment:
Great you used existing helper!
If you go deeper into its implementation, you'll see that there is detailed validation of each element of passed Array
```suggestion
    auto shape = js_to_cpp<ov::Shape>(info, 0);
```